js/components/Editor.js

```diff
@@ -50,36 +53,73 @@ const Editor = React.createClass({
   },
 
   onGutterClick(cm, line, gutter, ev) {
-    let info = cm.lineInfo(line);
-    cm.setGutterMarker(
-      line,
-      "breakpoints",
-      info.gutterMarkers ? null : makeMarker()
-    );
-
```

Comment:
breakpoint gutters are now added as part of the `componentWillReceiveProps` lifecycle so that breakpoints are persisted as selected source changes. It is also more uni-directional 😉